File: leetcode/0062-unique-paths/solution.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        arr= [[0]*n for _ in range(m)]
        arr[0][0]=1
        for i in range(m):
            for j in range(n):
                if i == j ==0:
                    continue
                val=0
                if i>0:
                    val+=arr[i-1][j]
                if j>0:
                    val+=arr[i][j-1]
                arr[i][j]=val
        print(arr)
        return arr[m-1][n-1]

        

File: leetcode/0062-unique-paths/solution.ts
function uniquePaths(m: number, n: number): number {
    let dp :number [][]=[]

    for(let i=0;i<m;i++){
        dp[i]=[]
        for(let j=0;j<n;j++){
           if(i===0||j===0){
            dp[i][j] =1
           }

           else{
            let val = dp[i-1][j]
            val+=dp[i][j-1]
            dp[i][j] = val
           }
        }
    }
    return dp[m-1][n-1]
};
